### PR TITLE
Adding tests and porting over a few fixes from DokuWiki

### PR DESCRIPTION
**Pull request by splitbrain**
We want to replace our own copy of the IXR library with a composer dependency to your release. While working on it I came across a few smaller things we had fixed over the years that are not in your library, yet.

This pull request,

* adds the phpunit framework as a dev dependency
* adds some first tests (ported over from DokuWiki)
* fixes a bug with string handling
* make the date parsing more relaxed

Because all of these somewhat depend on each other I submit them as one pull request instead of multiple.

If you're interested, I could also contribute:

* automatic test running via github actions
* doc block updates
* automatic api doc generation via github actions

### CODE REVIEW

File: src/Message/Message.php [hunk 1]
```diff
@@ -150,7 +150,7 @@ public function tagClose($parser, $tag)
                 $valueFlag = true;
                 break;
             case 'string':
-                $value = (string)trim($this->_currentTagContents);
+                $value = (string)($this->_currentTagContents);
```
[the remark this answers is not in the document]
Comment (splitbrain) [reply src/Message/Message.php:153]: FYI the original DokuWiki bug report that led to this change is here: https://bugs.dokuwiki.org/1824.html

**Comment by tenzap** [comment]
Hi all & @kissifrot  , any news on this please? We are trying to get this into a Debian package and having the tests would be really nice.

**Comment by tenzap** [comment]
Hi, thanks for merging. However, the tests do actually fail. Any idea @kissifrot , @splitbrain ?
```
There were 2 failures:

1) IXR\tests\Message\ixr_library_ixr_message_test::testStringValue
Failed asserting that two arrays are equal.
--- Expected
+++ Actual
@@ @@
 Array (
-    0 => 'change'
+    0 => ' change  '
 )

/mnt/packages/git_repos/dpkg/php-kissifrot-php-ixr/tests/Message/MessageTest.php:50

2) IXR\tests\Message\ixr_library_ixr_message_test::testStruct
Failed asserting that two arrays are equal.
--- Expected
+++ Actual
@@ @@
 Array (
     0 => 'start'
-    1 => 'test text'
+    1 => 'test text   '
     2 => Array (...)
 )

/mnt/packages/git_repos/dpkg/php-kissifrot-php-ixr/tests/Message/MessageTest.php:107

ERRORS!
Tests: 17, Assertions: 16, Errors: 9, Failures: 2.
```

**Comment by tenzap** [comment]
I also have 9 errors:
```
There were 9 errors:

1) IXR\tests\DataType\DateTest::testParseIso with data set #0 ('2010-08-17T09:23:14', 1282036994)
Error: Call to a member function format() on bool

/usr/share/php/IXR/DataType/Date.php:69
/mnt/packages/git_repos/dpkg/php-kissifrot-php-ixr/tests/DataType/DateTest.php:50

2) IXR\tests\DataType\DateTest::testParseIso with data set #2 ('2010-08-17 09:23:14', 1282036994)
Error: Call to a member function format() on bool

/usr/share/php/IXR/DataType/Date.php:69
/mnt/packages/git_repos/dpkg/php-kissifrot-php-ixr/tests/DataType/DateTest.php:50

3) IXR\tests\DataType\DateTest::testParseIso with data set #3 ('20100817 09:23:14', 1282036994)
Error: Call to a member function format() on bool

/usr/share/php/IXR/DataType/Date.php:69
/mnt/packages/git_repos/dpkg/php-kissifrot-php-ixr/tests/DataType/DateTest.php:50

4) IXR\tests\DataType\DateTest::testParseIso with data set #6 ('2010-08-17 09:23:14+0000', 1282036994)
Error: Call to a member function format() on bool

/usr/share/php/IXR/DataType/Date.php:69
/mnt/packages/git_repos/dpkg/php-kissifrot-php-ixr/tests/DataType/DateTest.php:50

5) IXR\tests\DataType\DateTest::testParseIso with data set #7 ('2010-08-17 09:23:14+00:00', 1282036994)
Error: Call to a member function format() on bool

/usr/share/php/IXR/DataType/Date.php:69
/mnt/packages/git_repos/dpkg/php-kissifrot-php-ixr/tests/DataType/DateTest.php:50

6) IXR\tests\DataType\DateTest::testParseIso with data set #8 ('2010-08-17 12:23:14+03:00', 1282036994)
Error: Call to a member function format() on bool

/usr/share/php/IXR/DataType/Date.php:69
/mnt/packages/git_repos/dpkg/php-kissifrot-php-ixr/tests/DataType/DateTest.php:50

7) IXR\tests\DataType\DateTest::testParseIso with data set #9 ('2010-08-17T09:23', 1282036980)
Error: Call to a member function format() on bool

/usr/share/php/IXR/DataType/Date.php:69
/mnt/packages/git_repos/dpkg/php-kissifrot-php-ixr/tests/DataType/DateTest.php:50

8) IXR\tests\DataType\DateTest::testParseIso with data set #10 ('20100817T09:23', 1282036980)
Error: Call to a member function format() on bool

/usr/share/php/IXR/DataType/Date.php:69
/mnt/packages/git_repos/dpkg/php-kissifrot-php-ixr/tests/DataType/DateTest.php:50

9) IXR\tests\DataType\DateTest::testParseIso with data set #11 ('2010-08-17', 1282003200)
Error: Call to a member function format() on bool

/usr/share/php/IXR/DataType/Date.php:69
/mnt/packages/git_repos/dpkg/php-kissifrot-php-ixr/tests/DataType/DateTest.php:50


```

**Comment by tenzap** [comment]
Forget it, it's a packaging error. I sorted it out. Sorry.

**Pull request merged**